packages/functional/actions/checkout-actions.ts

```diff
@@ -0,0 +1,56 @@
+import { faker } from '@faker-js/faker';
+import { expect, Page } from '@playwright/test';
+
+import { CheckoutPage } from '../pages/checkout-page';
+
+const testUser = faker.person.firstName();
+
+export async function fillGuestCustomerEmailAndShipping(page: Page) {
+  await page.locator(CheckoutPage.GUEST_CUSTOMER_EMAIL).click();
+  await page.locator(CheckoutPage.GUEST_CUSTOMER_EMAIL).fill(faker.internet.email());
```

Comment:
I was aiming for writing I18N tests for shipping related tests in the near future and I'm not sure if we use test based identifiers for such cases. I agree that in majoirity of cases, we should be testing like an end user but this might be a special case considering for I18N testing.

Reply:
You'd do something like this:
```
await page.goto('https://catalyst.dev/es/checkout');
await page.getByRole('textbox', { name: 'correo electrónico' });
```

I feel like those would be separate test cases, but we should aim to inline things as much as possible and then abstract whenever we see a consistent pattern. We will probably have a few tests that we fill out the shipping information differently than this method does, specifically i18n tests for other countries might include additional fields.

Reply:
I can move that into inline but ideally we should be able to run all tests against different locales, don't you think so? 

If we have to have do
`await page.getByRole('textbox', { name: 'correo electrónico' });` for Spanish and
`await page.getByRole('textbox', { name: 'email' });` for English then we end up writing way more tests don't we? 

Even if we can manage those strings in a JSON file for different locales and load those fixtures while running specific tests where we have to consider i18n, their maintenance will be difficult when we move fast. We will have to update across many places for different locales when a single text based selector changes in such cases.

Reply:
>we end up writing way more tests don't we?

Won't we still have write individual tests for each locale anyways?

>Even if we can manage those strings in a JSON file for different locales and load those fixtures while running specific tests where we have to consider i18n, their maintenance will be difficult when we move fast. We will have to update across many places for different locales when a single text based selector changes in such cases.

My big concern here is that we are abstracting this too early. We don't even have any tests written for i18n yet, so we don't even know what each test should do or if there are any edge-cases. I personally believe that we should just duplicate test logic into each test and then come back to abstracting it when we truly believe there is a need to do so.

Reply:
> Won't we still have write individual tests for each locale anyways?
Going to keep that discussion for next time when we work on locale tests :)

> My big concern here is that we are abstracting...
I updated the test to confine every action to stay within the test case since we aren't reusing it for now. I also updated almost all locators be identify as a end user. In two cases I had to rely on test locators since the ios rendering doesn't show role based selectors for those. Might be something we can fix later?